File: seminar_5/task_3/churilkin.artem/task3.cpp
#include<vector>


std::vector< std::vector <int> > pascal_triangle(int row, int col) {
    if (col == 0 or row == col) return {{1}};
    int n = 6;
    std::vector< std::vector <int> > dp;
    for (int i = 1; i <= n; i++){
        std::vector<int> tmp;
        for(int j = 1; j<= i; j++){ tmp.push_back(1); }
        dp.push_back(tmp);
    }
    for (row = 1; row < n; ++row) {
        for (col = 1; col < row; col++){
            dp[row][col] = dp[row-1][col-1] + dp[row-1][col];
        }
    }
    
    return dp;
}

int main() {
    return 0;
}

File: seminar_5/task_3/Khamitov.Islam/task_3.py
# Задача 3
# Нарисовать треугольник Паскаля 

def pascal_triangle(rows):
    dp = []
    
    for i in range(rows):
        row = [1]
        
        if i > 0:
            prev_row = dp[i - 1]
            for j in range(1, i):
                row.append(prev_row[j-1] + prev_row[j])
            row.append(1)

        dp.append(row)
    return dp

# example
pascal_triangle = pascal_triangle(6)
for row in pascal_triangle:
    print(row)